src/Sarif/Writers/SarifLogger.cs

```diff
@@ -447,12 +447,16 @@ private void CaptureArtifact(ArtifactLocation fileLocation)
                 catch (ArgumentException) { } // Unrecognized encoding name
             }
 
+            HashData hashData = null;
+            AnalysisTargetToHashDataMap?.TryGetValue(fileLocation.Uri.OriginalString, out hashData);
```

Comment:
>AnalysisTargetToHashDataMap?.TryGetValue [](http://example.com/codeflow?start=12&length=40)

if we have the hash, let's re-use it.
this will prevent us from hashing the same file twice